pkg/containerd/resolver/resolver.go

```diff
@@ -501,7 +501,7 @@ func (r *dockerBase) fetchTokenWithOAuth(ctx context.Context, to tokenOptions) (
 
 	// Registries without support for POST may return 404 for POST /v2/token.
 	// As of September 2017, GCR is known to return 404.
-	if (resp.StatusCode == 405 && r.username != "") || resp.StatusCode == 404 {
+	if (resp.StatusCode == 405 && r.username != "") || resp.StatusCode == 404 || resp.StatusCode == 401 {
```

Comment:
should this be if r.username != "" && resp.StatusCode == 401 ?

Reply:
can you dump the resp ? It should have some sort of challenge in the header on why it failed coz 401 auth related header.

Reply:
This just ports containerd/containerd#2171 to here.